File: java/src/Block/FlowControlBlock.java
/**
 * This file copyright (c) 2015-2016, Jing Fan, Adalbert Gerald Soosai Raj, and Jignesh M. Patel
 *
 * See the file CREDITS.txt in the root directory for details.
 **/

package Block;

/**
 * @brief The SQL block for flow control.
 */
public class FlowControlBlock extends Block {
    /**
     * @brief Constructor.
     * @param stage The stage string.
     * @param indent The indent level.
     * @param flowControl The flow control condition.
     * @param lhs The SQL block in if clause.
     * @param rhs The SQL block in else clause.
     */
  public FlowControlBlock(String stage,
                          int indent,
                          String flowControl,
                          String lhs,
                          String rhs) {
    super(stage, indent);
    this.append("IF (" + flowControl + ")");
    this.append("BEGIN");
    this.concat(lhs);
    this.append("END");
    this.append("ELSE");
    this.append("BEGIN");
    this.concat(rhs);
    this.append("END");
    this.sql = this.sb.toString();
  }

}

File: Blocks/FlowControlBlock.py
from Block import Block
class FlowControlBlock(Block):
    '''
    generates sql for the if..then..else block
    '''
    
    def __init__(self,stage,indent,flowControl,lhs,rhs):
        '''
        generates sql statement for if block
        @param stage: stage of the block
        @type stage: string
        @param indent: indent level
        @type indent: int
        @param flowControl: condition of the if statement
        @type flowControl: string
        @param lhs: sql to be executed if condition is true
        @type lhs: string
        @param rhs: sql to be executed if condition is false
        @type rhs: string
        '''
        super(FlowControlBlock,self).__init__(stage,indent)
        self.append("IF (" + flowControl + ")")
        self.append("THEN")
        self.concat(lhs)
        self.append("ELSE")
        self.concat(rhs)
        self.append("END IF;")
        self.sql = self.sb
        